stdlib/public/stubs/GlobalObjects.cpp

```diff
@@ -106,35 +106,33 @@ swift::_SwiftEmptySetStorage swift::_swiftEmptySetStorage = {
   0 // int entries; (zero'd bits)
 };
 
-static swift::_SwiftHashingSeed initializeHashingSeed() {
+static swift::_SwiftHashingParameters initializeHashingParameters() {
+  // Setting the environment variable SWIFT_DETERMINISTIC_HASHING to "1"
+  // disables randomized hash seeding. This is useful in cases we need to ensure
+  // results are repeatable, e.g., in certain test environments.  (Note that
+  // even if the seed override is enabled, hash values aren't guaranteed to
+  // remain stable across even minor stdlib releases.)
+  auto determinism = getenv("SWIFT_DETERMINISTIC_HASHING");
+  if (determinism && 0 == strcmp(determinism, "1")) {
+    return { 0, 0, true };
+  }
 #if defined(__APPLE__)
   // Use arc4random if available.
-  swift::_SwiftHashingSeed seed = { 0, 0 };
-  arc4random_buf(&seed, sizeof(seed));
-  return seed;
+  __swift_uint64_t seed0 = 0, seed1 = 0;
+  arc4random_buf(&seed0, sizeof(seed0));
+  arc4random_buf(&seed1, sizeof(seed1));
```

Comment:
Out of curiosity, is 2 calls equivalent to one? Is 2 calls better?

Reply:
It's equivalent for randomness; it may be *ever so slightly* slower. (No extra syscalls, just some potential RNG bookkeeping/call overhead.) The reason I went this way is that this seems cleaner than filling a separate 16-byte buffer and copying it into the result.